internal/server/server.go

```diff
@@ -203,7 +205,8 @@ func (h *Handler) Predict(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, "invalid procedure_source_url", http.StatusBadRequest)
 			return
 		}
-		srcDir := u.Path
+		hash := md5.Sum([]byte(u.Path))
```

Comment:
We use SHA256 in most places. Do that for consistency? Performance is not an issue here.

Reply:
We were assuming `req.ProcedureSourceURL` was a `file:///<path>` type local path that already exists.
But here we should probably dome something like:
```
var srcDir string
if u.Scheme == "file" {
    srcDir = u.Path
}
if u.Scheme == "http" || u.Schema == "https" {
    sha := <sha sum of URL>
    srcDir = path.Join(os.TempDir(), "procedure-" + sha)
    os.MkdirAll(srcDir)
    <pget -x URL srcDir>
}
```

Reply:
`pget -x <URL> <dir>` creates the `<dir>` so don't even need `os.MkdirAll`